tsfc/fem.py

```diff
@@ -699,4 +699,17 @@ def compile_ufl(expression, interior_facet=False, point_sum=False, **kwargs):
     result = map_expr_dags(context.translator, expressions)
     if point_sum:
         result = [gem.index_sum(expr, context.point_indices) for expr in result]
+    # Check that any indices from the PointSet have made it into the resulting expression
+    for i in range(len(result)):
+        expansion_indices = ()
+        for index in context.point_indices:
+            if index not in result[i].free_indices:
+                expansion_indices = expansion_indices + (index,)
+        if expansion_indices != ():
+            # blow up shape to include expansion_indices
+            deltas = gem.Delta(expansion_indices[0], expansion_indices[0])
+            # Use deltas to increase the possible rank of our expression
+            for index in expansion_indices[1:]:
+                deltas *= gem.Delta(index, index)
+            result[i] = gem.Indexed(gem.ComponentTensor(result[i] * deltas, expansion_indices), expansion_indices)
```

Comment:
`functools.reduce(gem.Product, (gem.Delta(i, i) for i in indices), ())`

no need for the guard because `()` is the identity for Indexed and CT

Reply:
Apparently not? This fails with 
```
../tsfc/tsfc/fem.py:708: in compile_ufl
    deltas = functools.reduce(gem.Product, (gem.Delta(j, j) for j in expansion_indices), ())
../tsfc/gem/gem.py:48: in __call__
    obj = super(NodeMeta, self).__call__(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'gem.gem.Product'>, a = ()
b = Delta(Index(2), Index(2))

    def __new__(cls, a, b):
>       assert not a.shape
E       AttributeError: 'tuple' object has no attribute 'shape'
```
seems I do need to guard

Reply:
Oh sorry, `()` is not closed in gem.

```
deltas = () if not deltas else functools.reduce(gem.Product, deltas)
```

Reply:
Don't quite follow your second suggestion here I'm afraid

Reply:
added a comment.